crates/recent_projects/src/recent_projects.rs

```diff
@@ -431,20 +429,7 @@ impl RecentProjectsDelegate {
                     .recent_workspaces_on_disk()
                     .await
                     .unwrap_or_default();
-                let mut unique_added_paths = HashMap::default();
-                for (id, workspace) in &workspaces {
-                    for path in workspace.paths().iter() {
-                        unique_added_paths.insert(path.clone(), id);
-                    }
-                }
-                let updated_paths = unique_added_paths
-                    .into_iter()
-                    .sorted_by_key(|(_, id)| *id)
-                    .map(|(path, _)| path)
-                    .collect::<Vec<_>>();
                 this.update(&mut cx, move |picker, cx| {
-                    cx.clear_recent_documents();
```

Comment:
This is a design decision of a kind: with this removal, we abandon the original idea of mimicking recent projects (and their order) in the menu.

I like how the code is simpler with it, so let's move on but then there's no real use case for `clear_recent_documents` in this model? Let's remove it entirely if it's true, both from the `cx` and the platform code.

Reply:
I think Zed doesn't have a use for it, but is it possible some other GPUI user down the line would want such a function? Electron exposes such a function in their API, so it might be worth maintaining one for GPUI as well. BTW, since we're now only adding paths one by one, would it be better to change `add_recent_documents` to `add_recent_document` and accept a single `PathBuf` rather than an array? (this also aligns more with Electron/macOS native API)

Reply:
Definitely, let's have a single path instead (and maybe have a rustdoc that explains the macOS part that it's a stack-like list there).

I think we'd rather have a smallest possible API exposure of this feature, as it's not very common, and wait for the real need of the `clear_recent_documents` — and remove it for now.